nilearn/glm/first_level/first_level.py

```diff
@@ -944,17 +955,33 @@ def first_level_from_bids(dataset_path, task_label, space_label=None,
                      img_specs[0])
 
     # Infer subjects in dataset
-    sub_folders = glob.glob(os.path.join(derivatives_path, 'sub-*/'))
-    sub_labels = [os.path.basename(s[:-1]).split('-')[1] for s in sub_folders]
-    sub_labels = sorted(list(set(sub_labels)))
+    if sub_labels:
+        sub_folders_in = [os.path.join(derivatives_path, f"sub-{s}") for s in sub_labels]
+        
+        sub_labels_exist = []
+        sub_labels_missing = []
+        sub_folders = []
+        for i, sub_folder in enumerate(sub_folders_in):
+            if os.path.exists(sub_folder):
+                sub_labels_exist.append(sub_labels[i])
+                sub_folders.append(sub_folder)
+            else:
+                sub_labels_missing.append(sub_labels[i])
+        if sub_labels_missing:
+            warn('The following subject labels are not present in the dataset'
+                 ' and cannot be processed: %s' %sub_labels_missing)
```

Comment:
I think here you can get rid of `sub_labels_missing` and just put the warning under the else statement and the replace `%sub_labels_missing` with `% sub_labels[i]`